### PR TITLE
Add zonemap sum

### PR DESCRIPTION
**Pull request by w-zr**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #10856

## What this PR does / why we need it:

Add sum in zonemap for int8, int16, int32, int64, uint8, uint16, uint32, uint64, float32, float64 and types.Decimal64.
Sum is stored in zm[16:24].
If overflowed, the sum returns 0.
Fix bug in Decimal64.Add64()

### CODE REVIEW

**Review by zengyan1 — APPROVED** [review]

File: pkg/vm/engine/tae/index/zm.go [hunk 1]
```diff
@@ -90,6 +91,11 @@ func (zm ZM) String() string {
 	} else {
 		_, _ = b.WriteString(fmt.Sprintf("ZM(%s)%d[%v,%v]",
 			zm.GetType().String(), zm.GetScale(), zm.GetMin(), zm.GetMax()))
+		if zm.GetType().IsFloat() {
+			_, _ = b.WriteString(fmt.Sprintf("SUM:%v", zm.GetSum()))
+		} else if zm.GetType().IsInteger() || zm.GetType() == types.T_decimal64 {
+			_, _ = b.WriteString(fmt.Sprintf("SUM:%v", zm.GetSum().(types.Decimal128).Format(0)))
```
Comment (fengttt) [inline pkg/vm/engine/tae/index/zm.go:97]: Decimal128?

**Review by fengttt — COMMENTED** [review]
It is incomplete.

You should also have updated metadata_scan(), so that you can select it from SQL.
Then you can actually test it.    insert delete flush merge ...

File: pkg/vm/engine/tae/index/zm.go (hunk 1 above)
Comment (w-zr) [reply pkg/vm/engine/tae/index/zm.go:97]: golang does not support int128...

**Review by XuPeng-SH — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Comment by w-zr** [comment]
> It is incomplete.
> 
> You should also have updated metadata_scan(), so that you can select it from SQL. Then you can actually test it. insert delete flush merge ...

Updated metadata_scan().

**Review by m-schen — APPROVED** [review]

**Pull request merged**